Week_01/G20200343030529/LeetCode_1_529.java

```diff
@@ -0,0 +1,28 @@
+import java.util.HashMap;
+
+/*
+ * @lc app=leetcode.cn id=1 lang=java
+ *
+ * [1] 两数之和
+ */
+
+// @lc code=start
+class Solution {
+    public int[] twoSum(int[] nums, int target) {
+        int[] result = new int[2];
+        HashMap<Integer, Integer> map = new HashMap<>();
+        for (int i = 0; i < nums.length; i++) {
+            int another = target - nums[i];
+            if (map.containsKey(another)) {
+                result[0] = map.get(another);
+                result[1] = i;
+                return result;
```

Comment:
12行的`result`只会在这里用到。可以省掉那个变量，返回时再动态生成。
```suggestion
                return new int[]{i, map.get(another};
```